src/main/java/me/jellysquid/mods/lithium/mixin/gen/features/StructureFeatureMixin.java

```diff
@@ -0,0 +1,74 @@
+package me.jellysquid.mods.lithium.mixin.gen.features;
+
+import net.minecraft.structure.StructureStart;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.util.math.ChunkSectionPos;
+import net.minecraft.world.WorldView;
+import net.minecraft.world.chunk.Chunk;
+import net.minecraft.world.chunk.ChunkStatus;
+import net.minecraft.world.gen.ChunkRandom;
+import net.minecraft.world.gen.StructureAccessor;
+import net.minecraft.world.gen.chunk.StructureConfig;
+import net.minecraft.world.gen.feature.FeatureConfig;
+import net.minecraft.world.gen.feature.StructureFeature;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+
+@Mixin(StructureFeature.class)
+public class StructureFeatureMixin {
+
+    /**
+     * @reason Why generate an empty chunk to check for a structure if
+     * the chunk's biome cannot generate the structure anyway?
+     * Checking the biome first = SPEED!
+     * @author TelepathicGrunt
+     */
+    @Overwrite
```

Comment:
If it's a simple one line change, wouldn't it be better for this to be an injector or a redirector?

Reply:
Replying here for other see as well and offer ideas.

The issue is we are in 2 nested for loops and need to skip this entire chunk of code if the biome is invalid. i couldn't figure out how to make an inject mixin to do this but maybe someone has an idea
```
Chunk chunk = worldView.getChunk(chunkPos.x, chunkPos.z, ChunkStatus.STRUCTURE_STARTS);
                            StructureStart<?> structureStart = structureAccessor.getStructureStart(ChunkSectionPos.from(chunk.getPos(), 0), thisStructure, chunk);
                            if (structureStart != null && structureStart.hasChildren()) {
                                if (skipExistingChunks && structureStart.isInExistingChunk()) {
                                    structureStart.incrementReferences();
                                    return structureStart.getPos();
                                }

                                if (!skipExistingChunks) {
                                    return structureStart.getPos();
                                }
                            }
```

Reply:
Some ideas kicked around. hmm 
There might not be a better clean idea other than a overwrite. rip

![image](https://user-images.githubusercontent.com/40846040/88458987-caed8480-ce5f-11ea-9f9e-87c212336a42.png)